amazonsqs/amazonsqs_connector.bal

```diff
@@ -0,0 +1,193 @@
+// specific language governing permissions and limitations
+// under the License.
+
+import ballerina/crypto;
+import ballerina/encoding;
+import ballerina/http;
+import ballerina/internal;
+import ballerina/io;
+import ballerina/system;
+import ballerina/time;
+import wso2/amazoncommons;
+import ballerina/log;
+
+
+# Object to initialize the connection with Amazon SQS.
+#
+# + accessKey - The Amazon API access key
+# + secretKey - The Amazon API secret key
+# + region - The Amazon API Region
+# + acctNum - The account number of the SQS service
+public type Client client object {
+
+    http:Client clientEp;
+    string accessKey;
+    string secretKey;
+    string region;
+    string acctNum;
+    string host;
+
+    public function __init(Configuration config) {
+        self.accessKey = config.accessKey;
+        self.secretKey = config.secretKey;
+        self.acctNum = config.accountNumber;
+        self.region = config.region;
+        self.host = SQS_SERVICE_NAME + "." + self.region + "." + AMAZON_HOST;
+        self.clientEp = new("https://" + self.host);
+    }
+
+    public remote function createQueue(string queueName, map<string> attributes) returns string|error;
+
+    public remote function sendMessage(string messageBody, string queueResourcePath, map<string> attributes) returns OutboundMessage|error;
+
+    public remote function receiveMessage(string queueResourcePath, map<string> attributes) returns InboundMessage[]|error;
+
+    public remote function deleteMessage(string queueResourcePath, string receiptHandle) returns boolean|error;
+    
+};
+
+# Creates a new queue in SQS
+#
+# + queueName - Name of the queue to be created 
+# + attributes - Other attribute parameters 
+# + return - URL of the created queue
+public remote function Client.createQueue(string queueName, map<string> attributes) returns string|error{
+
+    string amzTarget = "AmazonSQSv20121105.CreateQueue";
+    string endpoint = "/";
+    string payload =  "";
+    payload = payload + "Action=CreateQueue";
+    payload = payload + "&Version=2012-11-05";
+    payload = payload + "&QueueName=" + queueName;
+    int attributeNumber = 1;
+    foreach var (k, v) in attributes {
+        payload = payload + "&Attribute." + attributeNumber + ".Name=" + k;
+        payload = payload + "&Attribute." + attributeNumber + ".Value=" + v;
+        attributeNumber = attributeNumber + 1;
+    }
+
+    http:Request request = check generatePOSTRequest(self.accessKey, self.secretKey, self.host, amzTarget, endpoint, self.region, payload);
+    var httpResponse = self.clientEp->post(endpoint, request);
+    json|error response = handleResponse(httpResponse);
+    if(response is json){
+        return jsonToCreatedQueueUrl(response);
+    } else {
+        return response;
+    }
+
+}
+
+# Send a new message to a SQS queue
+#
+# + messageBody - Message body string to be sent 
+# + queueResourcePath - Resource path to the queue from the host address 
```

Comment:
shall we give an example resource path as well to reduce confusion for the full URL?

Reply:
Replaced with _queueResourcePath_